frontend/src/components/DestructionListReviewer/DestructionListReviewer.tsx

```diff
@@ -31,51 +41,143 @@ export type DestructionListReviewerProps = {
 export function DestructionListReviewer({
   destructionList,
 }: DestructionListReviewerProps) {
-  const [modalState, setModalState] = useState<boolean>(false);
   const { state } = useNavigation();
   const revalidator = useRevalidator();
   const alert = useAlert();
+  const formDialog = useFormDialog();
   const reviewers = useReviewers();
+  const coReviewers = useCoReviewers();
+  const assignedCoReviewers = useDestructionListCoReviewers(destructionList);
   const user = useWhoAmI();
 
   /**
    * Gets called when the change is confirmed.
-   * @param _
-   * @param data
    */
-  const handleSubmit = (
-    _: FormEvent,
-    { reviewer, comment }: SerializedFormData,
-  ) => {
-    if (!modalState) {
-      return;
-    }
-    reassignDestructionList(destructionList.uuid, {
-      assignee: { user: Number(reviewer) },
-      comment: String(comment),
-    })
-      .then(revalidator.revalidate) // Reload the current route.
-      .catch(async (e) => {
+  const handleSubmit = (data: SerializedFormData) => {
+    const { coReviewer, reviewer, comment } = data as {
+      comment: string;
+      reviewer?: string;
+      coReviewer?: string[];
+    };
```

Comment:
I am not convinced that this should happen in the same form :thinking: 

Currently the "reassign" endpoint assumes that the new assignee has  the role "main_reviewer", so this will not create co-reviewers. 

I think that we can discuss this as part also of whether the main reviewer should be able to unassign themselves.

Reply:
Given that they both should be able to be changed (otherwise this is a non-issue), both endpoints are called separately, the coReviewers are updated in a separate call from the reviewer.